pkg/daemon/update.go

```diff
@@ -936,6 +936,24 @@ func (dn *Daemon) updateOnClusterBuild(oldConfig, newConfig *mcfgv1.MachineConfi
 		return err
 	}
 
+	// Determine the pool (worker or master) based on the node's labels
+	pool := ""
+	var ok bool
+	if dn.node != nil {
+		if _, ok = dn.node.Labels["node-role.kubernetes.io/worker"]; ok {
+			pool = "worker"
+		} else if _, ok = dn.node.Labels["node-role.kubernetes.io/master"]; ok {
+			pool = "master"
+		}
+	}
+
+	// Ensure the MCN spec is updated with the correct desiredConfig
+	err = upgrademonitor.GenerateAndApplyMachineConfigNodeSpec(dn.featureGatesAccessor, pool, dn.node, dn.mcfgClient)
+	if err != nil {
+		klog.Errorf("Error updating MCN spec for node %s: %v", dn.node.Name, err)
```

Comment:
question: any reason we are logging the error in addition to returning it? Why not just do a `return fmt.Errorf()`?